src/locale/helpers.ts

```diff
@@ -31,9 +32,37 @@ export function code3ToCode2Strict(lang: string): string | undefined {
   return undefined
 }
 
-export function codeToLanguageName(lang: string): string {
-  const lang2 = code3ToCode2(lang)
-  return LANGUAGES_MAP_CODE2[lang2]?.name || lang
+function getLocalizedLanguage(
+  langCode: string,
+  appLang: string,
+): string | undefined {
+  const allNames = new Intl.DisplayNames([appLang], {
+    type: 'language',
+    fallback: 'none',
+    languageDisplay: 'standard',
+  })
+  const translatedName = allNames.of(langCode)
+
+  if (translatedName) {
+    // force simple title case (as languages do not always start with an uppercase in Unicode data)
+    return translatedName[0].toLocaleUpperCase() + translatedName.slice(1)
+  }
+}
+
+export function languageName(language: Language, appLang: string): string {
+  // if Intl.DisplayNames is unavailable on the target, display the English name
+  if (!(Intl as any).DisplayNames) {
+    return language.name
+  }
+
+  return getLocalizedLanguage(language.code2, appLang) || language.name
+}
```

Comment:
Could this be rewritten as a hook, to avoid needing to pass in langPrefs everywhere?
e.g.
```
export function useLanguageName(language: Language): string {
  const {appLanguage} = useLangPrefs();
  ...
}
```

Reply:
As hooks shall not be called conditionally, an hypothetical `useLanguageName` hook could not be used as-is inside `.map` calls… where most of the existing calls to `codeToLanguageName` are done.

So yeah, I am not convinced we would gain readability by using a hook here!